### DISCUSSION TITLE
Find result indicators - Show find result indicators in the vertical scroll bar

### DISCUSSION THREAD

**Discussion by advice2020** [geany/geany#4008] [Ideas]
I did not come up with this idea, this is a pretty common feature used in many applications.
Suggesting to add bold horizontal colored lines into the vertical scroll bar to show indicators of where keyword search items are located in the current document.  The current instance of that keyword that is highlighted in the document, gets a different color horizontal bold line to indicate to the user where it is in the document.

This request goes well with the other feature request that I made recently which can be found here if anyone is interested #4007 

I think this one could be very useful, hopefully others agree.

Thank You

**Comment by elextr** [comment]
Yes, that could be useful, and showing error locations too.  It would need to be requested from Scintilla AFAICT since it provides the scrollbars.

**Comment by elextr** [comment]
But having any type of position relative information within a scrollbar seems to not be supported by GTK, so it probably is not possible.

**Comment by advice2020** [comment]
(Deleted my previous response)
Good idea about also using something like this for showing error locations.

Chromium based web browsers seem to support this feature.
Have not been able to find a clear answer about what toolkit is used.
I have even read that some of these browsers have option that let user choose which toolkit to use in settings.
If it uses GTK then this could help determine what that it is possible, but there is also the question of GTK2 3 4 used by Geany and Chromium.

@nyamatongwe
Hello again
If okay, here is one of the posts that I had mentioned.
Do you by any change know if something like this would be possible to achieve in Scintilla?
Or is the information that user "elextr" mentioned above about this not being possible because of GTK limitations preventing this from allowing this.
Thank You

**Reply by nyamatongwe** [reply]
I've never found any information on drawing in the scroll bar for GTK.

**Reply by advice2020** [reply]
@nyamatongwe 
Ahh that is unfortunate.
Thank you so much for taking the time to look into this post as well.

Anyone
I know Firefox based browsers do not offer this but I wonder how Chromium based web browsers achieve this?
I could have sworn I have also seen this feature in other programs but I cannot remember any at the moment (might have been in Windows applications).
Such a useful visual feature, wish more applications would be able to provide this.
Anyways, if anyone has any ideas about this please let me know.

Thank You

**Reply by ntrel** [reply]
> I know Firefox based browsers do not offer this

Mine does (Linux Mint 22):
<img width="312" height="380" alt="image" src="https://github.com/user-attachments/assets/dce766f1-7488-453c-9b1f-e8f7caea6529" />

**Reply by b4n** [reply]
My Firefox does as well.

Anyway, it seems to just work by drawing over it:
<details><summary>Sample Python code</summary>

```python
import math
import gi
gi.require_version('Gtk', '3.0')  # noqa
from gi.repository import Gtk


# Interesting piece
class MyScrollbar(Gtk.Scrollbar):
    def do_draw(self, cr):
        Gtk.Scrollbar.do_draw(self, cr)

        alloc = self.get_allocation()
        cr.set_line_width(1)
        cr.set_source_rgba(1, 0, 1, 0.5)
        for mark in (0.05, 0.2, 0.5, 0.55, 0.8, 0.9):
            cr.move_to(math.floor(alloc.width * mark) + 0.5, 0)
            cr.line_to(math.floor(alloc.width * mark) + 0.5, alloc.height)
            cr.stroke()


# Boilerplate follows
class Window(Gtk.ApplicationWindow):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)

        self.scroll = MyScrollbar(visible=True)
        self.add(self.scroll)


class App(Gtk.Application):
    def __init__(self, *args, application_id='test.ScrollIndicTest', **kwargs):
        super().__init__(*args, application_id=application_id, **kwargs)

    def do_activate(self):
        window = self.get_active_window()
        if not window:
            window = Window(application=self, default_width=600)
        window.present()


if __name__ == '__main__':
    App().run()
```
</details>

**Reply by advice2020** [reply]
Thank you guys for testing this out.
That is strange, when I tested this in Firefox I did not see these?
At I will test again.  Probably not but wonder if maybe will see if I change my theme.

I apologize, I am not familar with what to do with this Python code, if applied does this replicate this somehow in Geany?

So then could something like this be added into Geany in the future?
It was explained to me that Geany does not generate the amount of keyword instances found so maybe that might be a separate reason this cannot be achieve, but it was also said above that it might not be possible in GTK?
Firefox is using GTK? maybe newer version?
Really nice feature for Find tool but also as user "elextr" mentioned also could be used for showing error locations.
